Documentation for mod_auth/forms.py

pranavsid98 opened the pull request:
Documentation for mod_auth/forms.py

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -33,12 +38,25 @@ def _email_not_in_use(form, field):
+	'''
+	Form for Login
+
+	attr email: user's email
```
This documentation doesn't provide any useful information (attr) to the user, and for an IDE it's not needed either.

pranavsid98 replied:
Did you mean that this particular Login Form didnt require documentation? Or does the mod_auth/forms.py file not require documentation as a whole?
Thanks

canihavesomecoffee replied:
A class description can be added, but the attr lines are of no use. A reader of the code can see the attributes right below the comment, and an IDE doesn't need them to show the attributes for auto-completion.

pranavsid98 commented:
Requested changes made

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -7,13 +7,18 @@
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    '''
+		Check if a user already exists with this name
```
It's even worse now.

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -7,13 +7,18 @@
 
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    '''
```
Double quotes please...

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -7,13 +7,18 @@
 def valid_password(form, field):
+	'''
+	Function to check for validity of a password
+	'''
```
Where's the parameter explanations?

pranavsid98 commented:
Is it better now? Changes made.

satyammittal commented on mod_auth/forms.py:
```diff
@@ -7,18 +7,24 @@
-    '''
-		Check if a user already exists with this name
-	'''
+    """
```
Please Check Indentation.

pranavsid98 commented:
@satyammittal 
I was following that layout initially. But, conforming to PEP 257, I changed it to double quotes. I changed the indentation. Any other suggestions?

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -28,7 +34,10 @@ def valid_password(form, field):
 
 
 def email_not_in_use(has_user_field=False):
-    def _email_not_in_use(form, field):
```
Why is this line removed? It's quite essential....

pranavsid98 replied:
Oh damn. My bad. Sorry. Didnt notice that I removed it.

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -7,13 +7,24 @@
 
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    	"""
```
Alignment is still not correct... Compare to the line with the comment that's gone now...

pranavsid98 replied:
Im not really sure why this alignment error is happening. I gave it exactly one tab space

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -7,13 +7,24 @@
+    	Check if a user already exists with this name
+
+	:param field: The data value for the 'name' inserted by new User
+	:type field.data : str
```
I'd rather see the type of field than field.data

canihavesomecoffee commented on mod_auth/forms.py:
```diff
@@ -23,7 +34,10 @@ def valid_password(form, field):
-    def _email_not_in_use(form, field):
+	"""
+	Function to check if the passed email is already in use.
+ 
```
Missing parameter

canihavesomecoffee commented:
And the layout of the mailer is factually pretty ok, but I missed out on the ' vs ". It definitely must be double quotes.

pranavsid98 commented:
Requested changes made

canihavesomecoffee commented:
It's fine, apart from the alignment. I'll correct that myself :)
